### ISSUE TITLE
Flexibility to skip empty json files in order to generate cucumber report

### ISSUE THREAD

**Issue by abhinandanbaheti** [damianszczepanik/cucumber-reporting#780]
Hi Damian,

In one my test cases execution flow, I am facing an issue that out of 200+ cucumber.json files ( which are created as a part of test results ), If few of the files are empty due to some test execution failures or runtime issues, then in that case the below function throws validation exception in the first json file coming in the array of list, as its unable to parse the empty json file, which in turn fails to generate cucumber report. Therefore, that makes it quite difficult to locate which are all the cucumber.json files are empty. 

/** ReportParser.java **/
 public List<Feature> parseJsonFiles(List<String> jsonFiles) {
/**/
    }

Therefore I was wondering, if we can provide a flag e.g. <skipjson> in the <classification> section of maven cucumber reporting plugin, which would give user the flexibility of skipping a empty cucumber.json files, while parsing all json files list, and generating report.

Please let me know, if you believe if this sounds feasible or not, or you have any better suggestion to address my problem ? I would be to excited to contribute here.

Regards,
Abhinandan

**Comment by damianszczepanik** [comment]
How would you define empty JSON?

**Comment by abhinandanbaheti** [comment]
Empty Cucumber.json file is basically a file with no content. And this empty file gets created if a environment in which the tests are running suddenly crashes, and test execution stops abruptly.

**Comment by damianszczepanik** [comment]
Well, JSON reports contains array of the feature so I expect the empty means empty array so ```[]```
Empty file should be treated as invalid JSON file rather empty

**Comment by damianszczepanik** [comment]
If you try https://jsonlint.com/ you will discover that no content is not valid JSON file

**Comment by abhinandanbaheti** [comment]
I completely agree with your point but my concern is, suppose if that invalid json file gets generated for few tests, due to test crashes, then this will basically cause maven cucumber reporting to fail at report parser. 

And User will not have any option to ignore these invalid json files in order to generate report through the plugin. He will have to manually delete those invalid json, and run the plugin again. 

My point was to basically provide a flag to user e.g. <skipInvalidJson>, which may provide flexibility to skip these invalid json files and successfully generate report, even though some of the tests failed.

**Comment by damianszczepanik** [comment]
There is discussion what are the rules for the cucumber report so invalid JSON files can be skipped so the report is not marked as failed. Can you add your point of view and wishes?
@calebmfreed @FastDev88 @abhinandanbaheti @baernhaut @Icaroto @aenky @imKT @ljzhangi @titudattaexiger  @AnuruddhaMk @anurag0510 @ptrthomas

**Comment by ptrthomas** [comment]
the next version of [Karate](https://github.com/intuit/karate) will not save invalid JSON anymore because we moved away from Cucumber-JVM - I'm not sure of the latest 4.X - but in Cucumber 1.2.X, the reporter would first create the JSON file and then stream content into it. IMO it is a problem that should be fixed in the framework creating the JSON. But - since there are so many folks out there who depend on Cucumber 1.2.X - it may make sense to have the flag.

**Comment by ptrthomas** [comment]
@FastDev88 I think ignoring only if the file is zero bytes and does not have a `{` within it is reasonable.

**Comment by damianszczepanik** [comment]
Feature is prepared, please review #794
